src/Entities/Player.cpp

```diff
@@ -426,12 +428,75 @@ void cPlayer::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 
 	Super::Tick(a_Dt, a_Chunk);
 
+	float Width;
+	float Height;
+	if (m_IsElytraFlying)
+	{
+		Width = 0.6F;
+		Height = 0.6F;
+	}
+	else if (m_bIsInBed)
+	{
+		Width = 0.2F;
+		Height = 0.2F;
+	}
+	else if (m_IsCrouched)
+	{
+		Width = 0.6F;
+		Height = 1.65F;
+	}
+	else
+	{
+		Width = 0.6F;
+		Height = 1.8F;
+	}
+
+	if ((Width != GetWidth()) || (Height != GetHeight()))
+	{
+		SetSize(Width, Height);
+		float EyeHeight = 1.62F;
+
+		if (IsInBed())
+		{
+			EyeHeight = 0.2F;
+		}
+		else if (IsCrouched())
+		{
+			EyeHeight -= 0.08F;
+		}
+		else if (IsElytraFlying())
+		{
+			EyeHeight = 0.4F;
+		}
```

Comment:
I don't like that the logic here is different from the logic above, could the two be combined so EyeHeight and Width/Height is guaranteed to be consistent. (the slight inefficiency from declaring the variable in advance is minor, and the compiler will probably optimise it out anyway)